src/main/java/org/dice_research/opal/launuts/polygons/LauNutsPolygonCentroid.java

```diff
@@ -0,0 +1,65 @@
+package org.dice_research.opal.launuts.polygons;
+
+import org.dice_research.opal.launuts.polygons.parser.*;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.json.simple.parser.ParseException;
+
+// https://howtodoinjava.com/library/json-simple-read-write-json-examples/
+
+/*
+ * Author: Amit Kumar
+ *
+ * This file implements the computation of nuts polygons centroid.
+ *
+ * Reusing parser code from NutsParser.java
+ *
+ * */
+
+public class LauNutsPolygonCentroid extends NutsParser {
+    org.dice_research.opal.launuts.polygons.MultiPolygon MultiPolygon = new org.dice_research.opal.launuts.polygons.MultiPolygon();
+    org.dice_research.opal.launuts.polygons.Point point = new org.dice_research.opal.launuts.polygons.Point();
+    Point pointCentroid = null;
+
+    public Point getNutsCenterPoint(String nutsCode) {
+        // This function calls getNutsPolygon in NutsParser and fetch the polygon found.
+        //Input: Nuts code
+        //Output: Centroid Point
+
+        try {
+
+            // Parsing nutsCode to getNutsPolygon to fetch polygon by iterating the JSON file.
+            MultiPolygon = getNutsPolygon(nutsCode);
+
+            // Calling function which will take polygon as input and compute the centroid
+            pointCentroid = computeCentroidMultiPolygon(MultiPolygon);
+
+            return pointCentroid;
+        } catch (Exception e) {
+
+        }
+        return null;
+    }
+
+    public Point computeCentroidMultiPolygon(MultiPolygon multiPolygon) throws IOException, ParseException, org.locationtech.jts.io.ParseException {
+        try {
```

Comment:
As discussed in the last meeting, we should not calculate centroid if a polygon has a hole. But your code is still calculating centroid if a polygon has holes. For example, Rastatt (DE124). https://ibb.co/4PB8B1q

We had also discussed that for a multipolygon there should be multiple centroids. But your code still calculates one centroid point for a multipolygon. For example, Hamburg (DE600) https://ibb.co/FH5nn4w